crawler/attribution/sw-source.js

```diff
@@ -0,0 +1,15 @@
+/* eslint-env serviceworker */
+
+self.addEventListener('install', (evt) => {
+    self.skipWaiting();
+});
+
+self.addEventListener('activate', (evt) => {
+    evt.waitUntil(self.clients.claim());
+});
+
+self.addEventListener('message', (event) => {
+    console.log(self.location.href, navigator.userAgent);
+    event.source.postMessage({ dave: self.location.href });
```

Comment:
Can we name this something meaningful? (or are we doing this elsewhere?)

Reply:
done, sorry :)